include/net/lwm2m.h

```diff
@@ -746,6 +753,29 @@ int lwm2m_engine_register_read_callback(char *pathstr,
 int lwm2m_engine_register_pre_write_callback(char *pathstr,
 					     lwm2m_engine_get_data_cb_t cb);
 
+/**
+ * @brief Set resource (instance) validation callback
+ *
+ * This callback is triggered before setting the value of a resource to the
+ * resource data buffer.
+ *
+ * The callback allows an LwM2M client or object to validate the data before
+ * writing and notify an error if the data should be discarded for any reason.
+ *
+ * @note All resources that have a validation callback registered are initially
+ *       decoded into a temporary validation buffer. Make sure that
+ *       ``CONFIG_LWM2M_ENGINE_VALIDATION_BUFFER_SIZE`` is large enough to
+ *       store each of the validated resources (individually).
+ *
+ * @param[in] pathstr LwM2M path string "obj/obj-inst/res(/res-inst)"
+ * @param[in] cb Validate resource data callback
+ *
+ * @return 0 if the data is accepted or a negative error code in case data is
```

Comment:
This doesn't look like describing return value of lwm2m_engine_register_validate_callback() (like should be done here), but rather of the validation callback.

Reply:
Indeed, will fix.